chain/chain/src/stateless_validation/chunk_endorsement.rs

```diff
@@ -117,19 +117,29 @@ pub fn validate_chunk_endorsements_in_block(
         // Validate the chunk endorsements bitmap (if present) in the block header against the endorsement signatures in the body.
         if let Some(endorsements_bitmap) = endorsements_bitmap {
             // Bitmap's length must be equal to the min bytes needed to encode one bit per validator assignment.
-            if endorsements_bitmap.len(shard_id).unwrap() == signatures.len().div_ceil(8) * 8 {
+            if endorsements_bitmap.len(shard_id).unwrap() != signatures.len().div_ceil(8) * 8 {
                 return Err(Error::InvalidChunkEndorsementBitmap(format!(
                     "Bitmap's length {} is inconsistent with the number of signatures {} for shard {} ",
                     endorsements_bitmap.len(shard_id).unwrap(), signatures.len(), shard_id,
                 )));
             }
+            // Bits in the bitmap must match the existence of signature for the corresponding validator in the body.
             for (bit, signature) in endorsements_bitmap.iter(shard_id).zip(signatures.iter()) {
                 if bit != signature.is_some() {
                     return Err(Error::InvalidChunkEndorsementBitmap(
                         format!("Chunk endorsement bit in header does not match endorsement in body. shard={}, bit={}, signature={}",
                         shard_id, bit, signature.is_some())));
                 }
             }
+            // All extra positions after the assignments must be left as false.
+            for value in endorsements_bitmap.iter(shard_id).skip(signatures.len()) {
+                assert_eq!(value, false);
```

Comment:
This will crash node if invalid bitmap is received, we should leave only the check below.

Reply:
ah yes, I left it while debugging. removed.